### PR TITLE
Add support for Silverstripe 5

### PR DESCRIPTION
**Pull request by MelissaWu-SS**
Upgrade the dependencies for supporting Silverstripe 5.

### CODE REVIEW

**Review by chrispenny — COMMENTED** [review]
Code changes look all good to me, and it won't even require a new major.

I haven't been able to test this; @MelissaWu-SS are you happy that it works as expected?

**Comment by MelissaWu-SS** [comment]
@chrispenny Thanks for the review! I've just tested it on project skeleton and another project, it works as expected on both.

**Review by chrispenny — APPROVED** [review]

**Comment by chrispenny** [comment]
Thanks, @MelissaWu-SS!

It doesn't look like I have write access to this module 😢 Perhaps Bernie does?

File: src/Executor.php [hunk 1]
```diff
@@ -9,7 +9,7 @@ class Executor
     /**
      * @var Purger
      */
-    private $purger;
+    private Purger $purger;
```
Comment (blueo) [inline src/Executor.php:12]: is this php 7 compatible? If not, we probably should bump the php version too

**Review by blueo — CHANGES_REQUESTED** [review]

File: src/Executor.php (hunk 1 above)
Comment (MelissaWu-SS) [reply src/Executor.php:12]: Nice spotting! I just checked it, looks like it is 7.4 compatible according to [this](https://wiki.php.net/rfc/typed_properties_v2), so we should bump the php version from ^7.3 to ^7.4?

File: composer.json [hunk 2]
```diff
@@ -10,11 +10,12 @@
     ],
     "license": "BSD-3-Clause",
     "require": {
-        "silverstripe/framework": "^4.0",
-        "silverstripe/admin": "^1.0"
+        "php": "^7.4 || ^8.1",
```
Comment (blueo) [inline composer.json]: sorry one more :) you can use cms4 on 8.0 so we should have ^8.0 :)

**Review by blueo — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (MelissaWu-SS) [reply composer.json]: Sweet, I've updated it🌟

**Review by blueo — APPROVED** [review]

**Pull request merged**